### PR TITLE
Fix missing comma

### PR DESCRIPTION
**Pull request by genos**
In #119, we noted that the missing comma in `qcs-api-client>=0.20.13<0.22.0` in `setup.py` was causing issues. With the changes here, the reported problem disappears:

```python
>>> import pennylane as qml
>>> qml.device("rigetti.qpu", device="Aspen-M-2")
<QPUDevice device (wires=80, shots=1000) at 0x132e949d0>
```

What's more, with `pip install -e '.[test]'` in a virtual environment, all unit tests pass (with some warnings).

### CODE REVIEW

**Comment by genos** [comment]
Sorry, I hadn't checked out the `.github/workflows` before making my PR! Hopefully 2583c76 helps.

File: setup.py [hunk 1]
```diff
@@ -1,60 +1,59 @@
...
-
-setup(classifiers=classifiers, **(info))
+#!/usr/bin/env python3
+from setuptools import setup
+
+with open("pennylane_rigetti/_version.py") as f:
+    version = f.readlines()[-1].split()[-1].strip("\"'")
+
+with open("./requirements.txt") as f:
+    requirements = [req.strip() for req in f.readlines()]
```
Comment (AlbertMitjans) [inline setup.py:8]: You deleted `requirements.txt`, thus this should raise an error

File: tests/test_qvm.py [hunk 2]
```diff
@@ -33,7 +33,6 @@
     "CZ 1 0\n"
     "MEASURE 1 ro[0]\n"
     "MEASURE 0 ro[1]\n"
-    "HALT\n"
```
Comment (AlbertMitjans) [inline tests/test_qvm.py:36]: What was this for?

File: requirements.txt [hunk 3]
```diff
@@ -1,5 +0,0 @@
-pyquil>=3.0.0,<4.0.0
```
Comment (AlbertMitjans) [inline requirements.txt:1]: Why was this file deleted?

**Comment by AlbertMitjans** [comment]
`pennylane` no longer supports `python3.7`. [I opened a PR](https://github.com/PennyLaneAI/pennylane-rigetti/pull/121) where I update the python version used in the docs and in the integration tests.

File: setup.py (hunk 1 above)
Comment (genos) [reply setup.py:8]: I added it back in but neglected to commit it! Terribly sorry, committing it now.

File: tests/test_qvm.py (hunk 2 above)
Comment (genos) [reply tests/test_qvm.py:36]: There's a test in `test_qvm.py` where the program is compared for equality. Running locally with `quilc -P -S` and `qvm -S`, there wasn't a `HALT` instruction and this test failed. If `quilc` is run without the protoquil `-P` flag, it's possible that this `HALT` instruction would be included; I'll put this back the way it was.

File: requirements.txt (hunk 3 above)
Comment (genos) [reply requirements.txt:1]: It was unused outside of CI, wasn't mentioned in the `README`, and had the potential to get out of sync with the requirements hardcoded in `setup.py`. I've since added it back, per my note above, but am using it in `setup.py` to ensure that there's a single source of truth on dependencies.

Comment (AlbertMitjans) [reply requirements.txt:1]: I agree! 💯

**Review by AlbertMitjans — APPROVED** [review]
Thanks for this! 🚀 

We'll need to merge my PR first.

**Comment by AlbertMitjans** [comment]
@MarquessV @genos do you have permission to review my PR? Otherwise I'll notify someone from the PL team.

**Comment by genos** [comment]
> @MarquessV @genos do you have permission to review my PR? Otherwise I'll notify someone from the PL team.

Sorry! I’m happy to take a look, but I lack GH permissions to give any official review.

**Review by AlbertMitjans — CHANGES_REQUESTED** [review]
Please add this fix in the changelog

**Review by AlbertMitjans — APPROVED** [review]
Thanks 🚀

File: setup.py [hunk 4]
```diff
@@ -1,11 +1,14 @@
...
+
 from setuptools import setup
 
 with open("pennylane_rigetti/_version.py") as f:
     version = f.readlines()[-1].split()[-1].strip("\"'")
 
-with open("./requirements.txt") as f:
-    requirements = [req.strip() for req in f.readlines()]
+
+requirements = ["pyquil>=3.0.0,<4.0.0", "qcs-api-client>=0.20.13<0.22.0", "pennylane>=0.18"]
```
Comment (genos) [inline setup.py:11]: This reinstates the problem the PR was trying to fix and ignores `requirements.txt`.

File: setup.py [hunk 5]
```diff
@@ -1,60 +1,60 @@
...
+import os
+import sys
+
+from setuptools import setup
+
+with open("pennylane_rigetti/_version.py") as f:
+    version = f.readlines()[-1].split()[-1].strip("\"'")
+
+
+requirements = ["pyquil>=3.0.0,<4.0.0", "qcs-api-client>=0.20.13<0.22.0", "pennylane>=0.18"]
```
Comment (AlbertMitjans) [inline setup.py:11]:
```suggestion
with open("./requirements.txt") as f:
    requirements = [req.strip() for req in f.readlines()]
```

File: setup.py (hunk 4 above)
Comment (AlbertMitjans) [reply setup.py]: Hi @genos! Thanks for noticing this. I must've missed it when fixing the merge conflict.

File: setup.py [hunk 6]
```diff
@@ -1,60 +1,61 @@
...
+#!/usr/bin/env python3
+import os
+import sys
+
+from setuptools import setup
+
+with open("pennylane_rigetti/_version.py") as f:
+    version = f.readlines()[-1].split()[-1].strip("\"'")
+
+
```
Comment (AlbertMitjans) [inline setup.py:10]:
```suggestion
```

File: setup.py (hunk 4 above)
Comment (AlbertMitjans) [reply setup.py]: Reverted the change. 👍

**Comment by genos** [comment]
@AlbertMitjans in the last 3 commits, I reinstated `requirements.txt`. What's more, I noticed that the GitHub actions were using different & out-of-date `quilc` versions, plus the test requirements were specified in multiple places; dca8a92 hopes to fix that.

File: .github/workflows/format.yml [hunk 7]
```diff
@@ -1,26 +1,26 @@
...
-        with:
-          python-version: 3.8
-
-      - name: Install dependencies
-        run: pip install black
-
-      - uses: actions/checkout@v2
-
-      - name: Run Black
-        run: black -l 100 pennylane_rigetti/ --check
```
Comment (AlbertMitjans) [inline .github/workflows/format.yml:26]: I don't know what is going on with github, but it is very difficult to review the PR when it says the whole file was changed 😅

Comment (genos) [reply .github/workflows/format.yml:26]: Apologies! There were DOS (aka CRLF) newlines in some (but not all) of these files, which my editor turned into just “\n”. No substantive changes in most, except for `quilc` versioning and `pip` commands.

**Comment by genos** [comment]
Perhaps also updating the `quilc` versions in testing was a bit beyond the scope of this PR 😂😬 Sorry! If tests fail again, I’ll set that versioning back and restrict the scope here to just `pip` usage.
